### PR TITLE
Convert pari_share() to str

### PR DESCRIPTION
**Pull request by jdemeyer**
Always use `str` instead of `bytes` for filenames.

@embray

### CODE REVIEW

**Comment by embray** [comment]
Looks good to me, thanks!

File: autogen/paths.py [hunk 1]
```diff
@@ -36,14 +36,17 @@ def pari_share():
...
         True
     """
     from subprocess import Popen, PIPE
     if not gppath:
         raise EnvironmentError("cannot find an installation of PARI/GP: make sure that the 'gp' program is in your $PATH")
     gp = Popen([gppath, "-f", "-q"], stdin=PIPE, stdout=PIPE)
     out = gp.communicate(b"print(default(datadir))")[0]
+    # Convert out to str if needed
+    if not isinstance(out, str):
+        out = out.decode("ascii", "surrogateescape")
```
Comment (embray) [inline autogen/paths.py]: Although I'm having second thoughts about the use of "ascii" here.  Maybe either `sys.getfilesystemencoding()` or at least utf-8?

**Comment by jdemeyer** [comment]
> Although I'm having second thoughts about the use of "ascii" here. Maybe either sys.getfilesystemencoding() or at least utf-8?

In any case, you're just guessing and I hate guessing.

Specifying `"ascii"` is being explicit that you have no clue really.

**Comment by embray** [comment]
`sys.getfilesystemencoding()` is at least an educated guess--it can still be wrong but usually only in some unusual cases.

**Comment by jdemeyer** [comment]
I see your point. And if Python assumes `sys.getfilesystemencoding()` when encoding filenames back as `bytes`, it should even be safe.

**Comment by embray** [comment]
I believe it does.  Though in fairness "ascii+surrogateescape" should be safe too--the whole point of "surrogateescape" is to accurate shuttle bytes to/from unicode even the wrong encoding was used (and thus the string is otherwise unreadable, by a human, as "text").

**Pull request merged**